Frameworks/AddressBook/ABRecord.mm

```diff
@@ -80,18 +152,404 @@ ABRecordType ABRecordGetRecordType(ABRecordRef record) {
+            address.locality = value[(__bridge NSString*)kABPersonAddressCityKey];
+            address.region = value[(__bridge NSString*)kABPersonAddressStateKey];
+            address.postalCode = value[(__bridge NSString*)kABPersonAddressZIPKey];
+            address.country = value[(__bridge NSString*)kABPersonAddressCountryKey];
+
+            [addresses addObject:address];
+        }
+    } else if (contactProperty == kABPersonDateProperty) {
+        _ABMultiValue* multiValue = (__bridge _ABMultiValue*)value;
+        if (!_checkType(kABDateTimePropertyType, multiValue, error)) {
+            return false;
+        }
+
+        // Filter importantDates so it only contains birthdays (since those should
+        // not be removed in this case.)
+        NSMutableArray* importantDates = person.contact.importantDates;
+        NSPredicate* predicate = [NSPredicate predicateWithBlock:^BOOL(id obj, NSDictionary* bindings) {
+            return ((WACContactDate*)obj).kind == WACContactDateKindBirthday;
+        }];
+
+        [importantDates filterUsingPredicate:predicate];
+
+        NSDictionary* dict = @{
+            ((__bridge NSString*)kABPersonAnniversaryLabel) : @(WACContactDateKindAnniversary),
+            ((__bridge NSString*)kABOtherLabel) : @(WACContactDateKindOther)
+        };
+
+        for (int i = 0; i < [multiValue getCount]; i++) {
+            NSString* label = (__bridge_transfer NSString*)[multiValue copyLabelAtIndex:i];
+            NSDate* value = (__bridge_transfer NSDate*)[multiValue copyValueAtIndex:i];
+            WACContactDate* date = [WACContactDate make];
+            date.kind = dict[label] == nil ? WACContactDateKindOther : [dict[label] integerValue];
+
+            unsigned int units = NSYearCalendarUnit | NSMonthCalendarUnit | NSDayCalendarUnit;
+            NSCalendar* calendar = [NSCalendar currentCalendar];
+            NSDateComponents* components = [calendar components:units fromDate:value];
+
+            date.year = @([components year]);
+            date.month = @([components month]);
+            date.day = @([components day]);
+
+            [importantDates addObject:date];
+        }
+    } else if (contactProperty == kABPersonPhoneProperty) {
+        _ABMultiValue* multiValue = (__bridge _ABMultiValue*)value;
+        if (!_checkType(kABStringPropertyType, multiValue, error)) {
+            return false;
+        }
+
+        // Ensure that none of the values violate the allowed length of Windows
+        // Contacts properties.
+        for (int i = 0; i < [multiValue getCount]; i++) {
+            NSString* value = (__bridge_transfer NSString*)[multiValue copyValueAtIndex:i];
+            if (!_checkLength(value, kABPersonPhoneLength, error, @"phone")) {
+                return false;
+            }
+        }
+
+        NSMutableArray* phones = person.contact.phones;
+        [phones removeAllObjects];
+
+        NSDictionary* dict = @{
```

Comment:
>NSDictionary [](start = 8, length = 12)

You can, if you want, save these dictionaries by making them static.

`static NSDictionary* abWACTypeMappingDictionary = [@{ ... } retain];`